File: matlab/lib/+Ice/AlreadyRegisteredException.m
classdef (Sealed) AlreadyRegisteredException < Ice.LocalException
    %ALREADYREGISTEREDEXCEPTION The exception that is thrown when you attempt to register an object more than once with
    %   the Ice runtime.
    %
    %   AlreadyRegisteredException Properties:
    %     kindOfObject - The kind of object that could not be removed.
    %     id - The ID (or name) of the object that is registered already.

    % Copyright (c) ZeroC, Inc.

    properties
        %KINDOFOBJECT The kind of object that could not be removed.
        %   character vector
        kindOfObject (1, :) char = ''

        %ID The ID (or name) of the object that is registered already.
        %   character vector
        id (1, :) char = ''
    end
    methods (Hidden)
        function obj = AlreadyRegisteredException(kindOfObject, id)
            % Convenience constructor without an errID or what message.
            if nargin == 0 % default constructor
                superArgs = {};
            else
                assert(nargin == 2, 'Invalid number of arguments');
                superArgs = {'Ice:AlreadyRegisteredException', sprintf('Another %s is already registered with ID ''%s''.', ...
                    kindOfObject, id)};
            end
            obj@Ice.LocalException(superArgs{:});
            if nargin > 0
                obj.kindOfObject = kindOfObject;
                obj.id = id;
            end
        end
    end
end

File: java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/AlreadyRegisteredException.java
// Copyright (c) ZeroC, Inc.

package com.zeroc.Ice;

/**
 * The exception that is thrown when you attempt to register an object more than once with the Ice runtime.
 */
public final class AlreadyRegisteredException extends LocalException {
    /**
     * Constructs an AlreadyRegisteredException.
     *
     * @param kindOfObject the kind of object that is already registered
     * @param id the ID (or name) of the object that is already registered
     */
    public AlreadyRegisteredException(String kindOfObject, String id) {
        super("Another " + kindOfObject + " is already registered with ID '" + id + "'.");
        this.kindOfObject = kindOfObject;
        this.id = id;
    }

    @Override
    public String ice_id() {
        return "::Ice::AlreadyRegisteredException";
    }

    /**
     * The kind of object that is already registered: "servant", "facet", "object", "default servant",
     * "servant locator", "plugin", "object adapter", "object adapter with router", "replica group".
     */
    public final String kindOfObject;

    /** The ID (or name) of the object that is already registered. */
    public final String id;

    private static final long serialVersionUID = 392587231034664196L;
}
